modules/tfm/tfm/boards/common/tfm_hal_platform.c

```diff
@@ -27,20 +31,22 @@ enum tfm_hal_status_t tfm_hal_platform_init(void)
 #if defined(TFM_PARTITION_CRYPTO)
 	int err;
 
+#if !defined(TFM_CRYPTO_RNG_MODULE_DISABLED)
 	/* Initialize the nrf_cc3xx runtime */
+#if defined(PSA_WANT_ALG_CTR_DRBG)
 	err = nrf_cc3xx_platform_init();
+#elif defined (PSA_WANT_ALG_HMAC_DRBG)
+	err = nrf_cc3xx_platform_init_hmac_drbg();
+#else
+	#error Please enable either PSA_WANT_ALG_CTR_DRBG or PSA_WANT_ALG_HMAC_DRBG
+#endif
+#else /* !defined(TFM_CRYPTO_RNG_MODULE_DISABLED) */
+	err = nrf_cc3xx_platform_init_no_rng();
+#endif
```

Comment:
To determine which nrf_cc3xx_platform_init* function to use we are using 2 PSA defines, and one TF-M define.

Is there any PSA define for a disabled RNG ?

Reply:
We need to clean this up with the same refactoring as mentioned here:

https://github.com/nrfconnect/sdk-nrfxlib/pull/645/files#r809971529

This should at the very least be done for the 2 PSA deifnes. It is less obvious to me if it should also apply to no_rng.